parser_test.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+from parser import ParagraphParser, BlockParser
+from block import TextBlock
+from bold import BoldBlock
+from italic_block import ItalicBlock
+from img import ImgBlock
+from link_block import LinkBlock
+from codeblock import CodeBlock
+from fenched_code_block import FenchedCodeBlock
+from strikethrough_block import StrikethroughBlock
+
+from header import Header
+from horizontal_rule import HorizontalRule
+from list_paragraph import OrderedList, UnorderedList
+from quote import QuoteParagraph
```

Comment:
The import will look better if we put elements in a package.

Reply:
> Re @jinglongwu
> Yeah I agree with this design and fix the assertion logic. However, I still think the logic `_invoke_parsers` in both ParagraphParser and BlockParser is not correct.
> For example: `BlockParser.parse('testing')`
> 
> ```python
>         for parser in self._parsers:
>             begin, end, element = parser(content)
>             # No matching parser for this content, begin can be -1
>             if begin != 0 and begin < start_index:
>                 # inside this if, start_index, end_index, final_element = -1, -1, None
>                 start_index, end_index, final_element = begin, end, element 
>         # at this time, start_index=-1
>         if start_index != 0 and start_index != len(content):
>             # error in content[0, -1]
>             child = TextBlock(content[0, start_index])
> ```

it fixed in #24